### PR TITLE
OpenStudio version required in projects, using version in ResStock develop for CI

### PR DESCRIPTION
**Pull request by nmerket**
## Pull Request Description

Updates CI to use whatever version of OpenStudio is specified in `national_baseline.yml`. This should future proof this from breaking again as it did when [this PR](https://github.com/NREL/resstock/pull/1225) merged in ResStock.

Also, I took the opportunity to require the `os_version` and `os_sha` in project files moving forward. We're including them in ResStock and they were added for ComStock, so it didn't make sense to maintain a default that nobody used anymore.

## Checklist

Not all may apply

- [x] Code changes (must work)
- [x] Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)
- [x] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/coverage.yml` as necessary.
- [x] All other unit and integration tests passing
- [x] Update validation for project config yaml file changes
- [x] Update existing documentation
- [x] ~~Run a small batch run on Kestrel/Eagle to make sure it all works if you made changes that will affect Kestrel/Eagle~~
- [x] Add to the changelog_dev.rst file and propose migration text in the pull request

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,8 +34,12 @@ jobs:
...
       - name: Download and Install OpenStudio
         run: |
-          wget -q https://github.com/NREL/OpenStudio/releases/download/v3.7.0/OpenStudio-3.7.0+d5269793f1-Ubuntu-22.04-x86_64.deb
-          sudo apt install -y ./OpenStudio-3.7.0+d5269793f1-Ubuntu-22.04-x86_64.deb
+          sudo snap install yq
+          export OS_VER=`yq '.os_version' resstock/project_national/national_baseline.yml`
+          export OS_SHA=`yq '.os_sha' resstock/project_national/national_baseline.yml`
+          export OS_INSTALLER_FILENAME="OpenStudio-${OS_VER}+${OS_SHA}-Ubuntu-22.04-x86_64.deb"
+          wget -q "https://github.com/NREL/OpenStudio/releases/download/v${OS_VER}/${OS_INSTALLER_FILENAME}"
+          sudo apt install -y "./${OS_INSTALLER_FILENAME}"
```
Comment (nmerket) [inline .github/workflows/ci.yml:42]: @joseph-robertson This should do it so they don't get out of sync anymore. Assuming the OpenStudio team doesn't change their naming and folder convention for their installers.

**Review by joseph-robertson — APPROVED** [review]
👍 !

**Comment by joseph-robertson** [comment]
@shorowit brought up the idea of (at least for residential / OS-HPXML) dropping the `os_version` and `os_sha` completely, and instead using the [OS-HPXML required version of OpenStudio](https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/version.rb) to determine the correct installer to use.

Edit: perhaps that would require OS-HPXML to store version and sha?

**Comment by nmerket** [comment]
> @shorowit brought up the idea of (at least for residential / OS-HPXML) dropping the `os_version` and `os_sha` completely, and instead using the [OS-HPXML required version of OpenStudio](https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/version.rb) to determine the correct installer to use.
> 
> Edit: perhaps that would require OS-HPXML to store version and sha?

Yeah, we would have to have the sha in that file too.

**Pull request merged**